### PR TITLE
[FLINK-20633][python] Add retry times to download avro

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request will add retry times to download avro*

## Brief change log

  - *Add retry times to download avro*

## Verifying this change

 - *Manual testing*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/pyflink/pyflink_gateway_server.py [hunk 1]
```diff
@@ -290,7 +291,23 @@ def launch_gateway_server_process(env, args):
         classpath = os.pathsep.join(
             [construct_flink_classpath(env), construct_hadoop_classpath(env)])
         if "FLINK_TESTING" in env:
-            download_apache_avro()
+            retry_times = 3
```
[the remark this answers is not in the document]
Comment (HuangXingBo) [reply flink-python/pyflink/pyflink_gateway_server.py:294]: This only downloads the avro package when the local test prepares the environment. It is completely invisible to the user. I don't think it is necessary to make the number of retries configurable. WDYT

File: flink-python/pyflink/pyflink_gateway_server.py [hunk 2]
```diff
@@ -290,7 +291,23 @@ def launch_gateway_server_process(env, args):
...
+            error = None
+            while retry_times > 0 and not status:
+                retry_times -= 1
+                try:
+                    download_apache_avro()
+                    status = 1
+                except CalledProcessError as e:
+                    status = 0
+                    error = e
+                    print("Retry {0} times to download avro.".format(retry_times))
```
Comment (SteNicholas) [inline flink-python/pyflink/pyflink_gateway_server.py:305]: Is this times of the output opposite?

Comment (HuangXingBo) [reply flink-python/pyflink/pyflink_gateway_server.py:305]: Print both messages should make it a little clearer. I'll change it

**Review by SteNicholas — APPROVED** [review]
LGTM. cc @dianfu

**Comment by gaoyunhaii** [comment]
@flinkbot run azure

**Pull request closed without merging**